src/screens/CreatePost/index.js

```diff
@@ -492,26 +494,31 @@ const CreatePost = () => {
               params: {refresh: true}
             });
           }
-
-          Analytics.logEvent('create_post', {
-            id: 6,
-            newpost_reach: renderLocationString(geoList[geoSelect]),
-            newpost_privacy: listPrivacy[0].label,
-            num_images: 0,
-            added_poll: isPollShown,
-            topics_added: listTopic,
-            anon: typeUser,
-            predicted_audience: audienceEstimations
+        } else {
+          showMessage({
+            message: StringConstant.createPostFailedGeneralError,
+            type: 'danger'
           });
+        }
 
-          return true;
+        Analytics.logEvent('create_post', {
+          id: 6,
+          newpost_reach: renderLocationString(geoList[geoSelect]),
+          newpost_privacy: listPrivacy[0].label,
+          num_images: 0,
+          added_poll: isPollShown,
+          topics_added: listTopic,
+          anon: typeUser,
+          predicted_audience: audienceEstimations
+        });
+      } catch (e) {
+        if (__DEV__) {
+          console.log('CreatePost : ', e);
+          console.warn('retryCount', retryCount);
         }
-        throw new Error('Post failed');
-      } catch (error) {
-        console.warn('retryCount', retryCount);
         if (retryCount >= maxRetries) {
           showMessage({
-            message: 'Post keeps failing. Please check your internet connection.',
+            message: 'Failed to post. Please check your internet connection.',
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling here is improved by providing more specific error messages based on the retry count. However, the same issue mentioned above applies here too. The `setLoadingPost(false)` statement won't be executed if an exception is thrown outside of the retry condition. You should also move this statement into a `finally` block.

```diff
         if (retryCount >= maxRetries) {
           showMessage({
            message: 'Failed to post. Please check your internet connection.',
             type: 'danger'
           });
-          setLoadingPost(false);
+        } finally {
+          setLoadingPost(false);
         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->